### PR TITLE
close writer

### PR DESCRIPTION
**Pull request by ChristinaXu2017**
# Description

Just fix a bug:  I forgot to close the file reader and writer at the end of the program,  so it causes some outputs missing at sometimes.  Here I put the writer and reader inside the try closure,  make sure they are automatically closed at the end. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Run a real big dataset. 

# Are WDL Updates Required?

not required at this pull request

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: qmule/src/org/qcmg/qmule/FixCutAdapter.java [hunk 1]
```diff
@@ -40,15 +40,11 @@ public static void main(final String[] argv) {
...
-				
-		@SuppressWarnings("resource")
-		FastqWriter writer =  OUTPUT == null ?
-				null : (new FastqWriterFactory()).newWriter(OUTPUT);
 		 			        
-        try{
+        try( BufferedReader reader = INPUT == null ? 
+				new BufferedReader(new InputStreamReader(System.in)) : IOUtil.openFileForBufferedReading(INPUT);
+        	FastqWriter writer =  OUTPUT == null ?
+				null : (new FastqWriterFactory()).newWriter(OUTPUT);) {
```
Comment (delocalizer) [inline qmule/src/org/qcmg/qmule/FixCutAdapter.java:47]: For symmetry you could create the stdout writer here too...
```
writer = OUTPUT == null ? new BufferedWriter(new OutputStreamWriter(System.out)) : ...
```
and then below you don't have to do the test if OUTPUT == null then `println`

Comment (delocalizer) [reply qmule/src/org/qcmg/qmule/FixCutAdapter.java:47]: or better:
```
FastqWriter writer = OUTPUT == null ? (new BasicFastqWriter(System.out)) : (new FastqWriterFactory()).newWriter(OUTPUT);)
```

Comment (ChristinaXu2017) [reply qmule/src/org/qcmg/qmule/FixCutAdapter.java]: sorry, I misunderstand it. now I changed to FastqWriter with BasicFastqWriter

**Review by delocalizer — APPROVED** [review]
looks good!

**Pull request merged**